src/main/java/frc/robot/Autonomous.java

```diff
@@ -0,0 +1,139 @@
+    // private Shooter shooter;
+    // private Intake intake;
+    // private Elevator elevator;
+    private BallPathImpl ballPath;
+    private double wheelCircumference = Math.pow((Math.PI*2), 2);
+    private double distance;
+    private double setPoint;
+    private double max = 1000;
+    private double min = 0;
+
+    private double kP = 0.00001;
+    private double kI = 0;
+    private double kD = 0;
+
+    private SparkMaxPIDController leftPIDController;
+    private SparkMaxPIDController rightPIDController;
+
+    private RelativeEncoder leftEncoder;
+    private RelativeEncoder rightEncoder;
+
+    private double maxSpeed = 1;
+    private double minSpeed = -1;
+    private double zRotation = 0;
+
+
+
+
+
+    public Autonomous(DriveImpl drivetrain, Shooter shooter, Intake intake, Elevator elevator){
+        this.drivetrain = drivetrain;
+        this.ballPath = new BallPathImpl(intake, elevator, shooter);
+        this.distance = 0;
+        this.setPoint = 0;
+
+        this.leftEncoder = drivetrain.leftEncoder;
+        this.rightEncoder = drivetrain.rightEncoder;
+        this.leftPIDController = drivetrain.leftSide.getPIDController();
+        this.rightPIDController = drivetrain.rightSide.getPIDController();
+
+
+        leftPIDController.setOutputRange(min, max);
+        rightPIDController.setOutputRange(min, max);
+
+    }
+
+    double calcDistance(RelativeEncoder encoder){
+
+        // Calculate distance from encoder ticks (in inches)
+
+        distance = calcTicks(encoder) * wheelCircumference;
+
+        return distance;
+    }
+
+    double calcTicks(RelativeEncoder encoder){
+        // Calculates ticks per revolution(shaft rotation)
+        double gearRatio = 8;
+        double ticksPer = encoder.getCountsPerRevolution();
+        double revs = encoder.getPosition();
+
+        double ticks = ((ticksPer * revs) / gearRatio);
+
+        return ticks;
+    }
+
+    double getCurrentPosition(RelativeEncoder encoder){
+        // return current encoder ticks for distance driven
+        return calcDistance(encoder) / wheelCircumference;
+    }
+
+    // void setSetPoint(double targetDistance){
+    //     setPoint = targetDistance / wheelCircumference;
+    // }
+
+    double convertIT(double distance){
+        // Convert Inches to Ticks
+        return distance / wheelCircumference;
+    }
+
+    double convertTI(double ticks){
+        // Convert Ticks to Inches
+        return ticks * wheelCircumference;
+    }
+
+    // double PIDCalc(PIDController pidController){
+    //     double output;
+    //     output = pidController.calculate(this.drivetrain.leftEncoder.getPosition());
+
+    //     return output;
+    // }
+
+    void positionPIDCalc(double target){
+        // PID for both sides
+        leftPIDController.setReference(target, ControlType.kPosition);
+        rightPIDController.setReference(-target, ControlType.kPosition);
+    }
+
+    void run(double targetPosition){
```

Comment:
The goal is to call this from within `Robot.autoRoutine`, like `autonomous.run(10 /* feet */);`, right?

Reply:
this is correct, I pushed without editing Robot.java but I will push my latest commit now where that is done.